### PR TITLE
AP_Landing: Support CCW Deepstalls

### PR DESCRIPTION
**Pull request by WickedShell**
This is a number of small deepstall enhancements:

- Support CCW deepstall loiters approaches
- Allow scaling aileron response up/down compared to the rudder when in deepstall. (Needed for some aircraft with ailerons)
- Rework the deepstall logging messages to improve the log analysis of deepstall approaches. (PIDL was only used for deepstall reasons, and a proper deepstall message made more sense then having the pair of messages)
- Cleanup some redundant code, remove k_aileron_with_input handling since it is no longer supported

SITL testing appears to be well behaved, however I'd like to get a real world test flight in before merging, as SITL testing can't test anything after the flare. Should be able to do the test flight tomorrow, but the code is ready for review (despite the WIP tag).

### CODE REVIEW

File: libraries/AP_Landing/AP_Landing.cpp [hunk 1]
```diff
@@ -580,6 +580,7 @@ void AP_Landing::log(void) const
         type_slope_log();
         break;
     case TYPE_DEEPSTALL:
+        deepstall.log();
```
Comment (magicrub) [inline libraries/AP_Landing/AP_Landing.cpp]: please add a break here.

Comment (WickedShell) [reply libraries/AP_Landing/AP_Landing.cpp]: Oops, fixed.

File: libraries/AP_Mission/AP_Mission.cpp [hunk 2]
```diff
@@ -604,6 +604,7 @@ MAV_MISSION_RESULT AP_Mission::mavlink_int_to_mission_cmd(const mavlink_mission_
     case MAV_CMD_NAV_LAND:                              // MAV ID: 21
         copy_location = true;
         cmd.p1 = packet.param1;                         // abort target altitude(m)  (plane only)
+        cmd.content.location.flags.loiter_ccw = is_negative(packet.param4); // yaw direction, (plane deepstall only)
```
Comment (magicrub) [inline libraries/AP_Mission/AP_Mission.cpp]: uh oh.. This PR was looking perfect until I saw this. You so sneaky!

Comment (WickedShell) [reply libraries/AP_Mission/AP_Mission.cpp]: Whats wrong with it? Param4 has an official meaning for yaw in the MAVLink spec :) If someone else wants to actually use the value we could stick it in a struct, but till then the flags field felt great :)

Comment (magicrub) [reply libraries/AP_Mission/AP_Mission.cpp]: hmm.. interesting. I didn't notice that at all! Thought you were trying to do something funny there. Although it's used out of spec still, a comment in common.xml would be required...
https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/common.xml#L703

**Comment by WickedShell** [comment]
Flight tested, removed the WIP tag.

**Comment by magicrub** [comment]
everything except the NAV_LAND param4 thing looks good. I'd prefer @tridge make an executive decision on it.

File: libraries/AP_Mission/AP_Mission.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_Mission/AP_Mission.cpp]: its a bit strange, but useful, so I think we can do this

**Comment by magicrub** [comment]
@tridge gave the thumbs up on the funny mavlink usage so merging.. Thank you!

**Pull request merged**